Sources/Swift/MetricKit/SentryMXManager.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+
+#if os(iOS) || os(macOS)
+
+#if canImport(MetricKit)
+import MetricKit
+#endif
+
+@available(iOS 14.0, macOS 12.0, *)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@objc public protocol SentryMXManagerDelegate: AnyObject {
+    
+    func didReceiveCrashDiagnostic(_ diagnostic: MXCrashDiagnostic, callStackTree: SentryMXCallStackTree, timeStampBegin: Date, timeStampEnd: Date)
+    
+    func didReceiveDiskWriteExceptionDiagnostic(_ diagnostic: MXDiskWriteExceptionDiagnostic, callStackTree: SentryMXCallStackTree, timeStampBegin: Date, timeStampEnd: Date)
+    
+    func didReceiveCpuExceptionDiagnostic(_ diagnostic: MXCPUExceptionDiagnostic, callStackTree: SentryMXCallStackTree, timeStampBegin: Date, timeStampEnd: Date)
+}
+
+@available(iOS 14.0, macOS 12.0, *)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@objc public class SentryMXManager: NSObject, MXMetricManagerSubscriber {
```

Comment:
l: You can user @objcMembers, and every function and property will already be objc compatible
```suggestion
@objcMembers public class SentryMXManager: NSObject, MXMetricManagerSubscriber {
```